index.js

```diff
@@ -48,16 +48,16 @@ function initialize() {
       preAgentTime
     )
 
-    // TODO: Update this check when Node v10 is deprecated.
-    if (psemver.satisfies('<10.0.0')) {
+    // TODO: Update this check when Node v12 is deprecated.
+    if (psemver.satisfies('<12.0.0')) {
```

Comment:
i feel like this series of checks could be simplified:
 * if !semver sastisfied: requires version based on engines
 * if >= 17.0.0: untested version

thoughts?

Reply:
I see what you mean and I agree. I'll rework this.